docs/source/api.rst

```diff
@@ -125,3 +125,67 @@ For example::
 
 
 
+Detailed API reference
+------------------------
+Module: :mod:`traitlets.traitlets`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. automodule:: traitlets.traitlets
+    :members:
+    :private-members:
```

Comment:
> Marking a method as private, conveys a strong enough signal to the consumer of the API, but we should not limit in addition her capability to understand the library.

This is not my experience.

> For instance, I had to read Configurable._load_config_files(filename, path) to see that the path arg in Configurable.load_config_file(filename.path) can be a list.

That is a failure of the documentation of the public API, not an indicator that the private implementation detail should be covered in the public documentation.

There are two *very* different goals of generated API docs:

1. how to use the library
2. how does the library work

Having a single document covering both causes confusion, and inevitably results in people confusing private APIs with public ones.

It is okay (debateable) to have generated documentation of the "internal implementation details" in the docs, but they should not be in the same place (especially not the same page) as the public, "how should I use this library" documentation. Given the general readability of Python, I find that the best balance is to only produce docs of the actual public APIs, and rely on reading source code for the internal implementation details.